### DISCUSSION TITLE
Index of ContactManifold triangle

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#992] [Q&A] [answered]
Bullet Physics had a mod at one point that would include the triangle index of the mesh that a ContactManifold was created from. This allowed you to check this like material properties of that particular triangle, etc. Think of a static world mesh where different materials cause different responses (i.e., a bouncy surface). Is the correct method in Jolt to split out the mesh by these different properties to actually create several different mesh objects and then handle the responses that way?

**Comment by jrouwe** [comment, marked as answer]
What you want is demo-ed here:

https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/FrictionPerTriangleTest.cpp

You can't get the triangle index (that's lost while optimizing the mesh) but you can assign materials to each triangle.
